packages/core/src/lib/sign-in-experience.test.ts

```diff
@@ -122,4 +130,50 @@ describe('validate sign-in methods', () => {
       );
     });
   });
+
+  test('should throw when the social connector IDs are empty and social sign-in method is enabled', () => {
+    expect(() => {
+      validateSignInMethods(
+        { ...mockSignInMethods, social: SignInMethodState.secondary },
+        [],
+        // @ts-expect-error-this-line
+        enabledConnectorInstances
+      );
+    }).toMatchError(new RequestError('sign_in_experiences.empty_social_connectors'));
+  });
+
+  describe('Selected social connectors must be enabled only when social sign-in method is enabled.', () => {
+    test('should not validate selected social connectors when social sign-in method is disabled', () => {
+      expect(() => {
+        validateSignInMethods(
+          { ...mockSignInMethods, social: SignInMethodState.disabled },
+          ['google', 'facebook'],
+          // @ts-expect-error-this-line
+          enabledConnectorInstances
+        );
+      }).not.toThrow();
+    });
+
+    test('should throw when some selected social connector are disabled and social sign-in method is enabled', () => {
+      expect(() => {
+        validateSignInMethods(
+          { ...mockSignInMethods, social: SignInMethodState.secondary },
+          ['google', 'facebook'],
+          // @ts-expect-error-this-line
+          enabledConnectorInstances
+        );
+      }).toMatchError(new RequestError('sign_in_experiences.invalid_social_connectors'));
+    });
+
+    test('should not throw when all selected social connectors are enabled and social sign-in method is enabled', () => {
+      expect(() => {
+        validateSignInMethods(
+          { ...mockSignInMethods, social: SignInMethodState.secondary },
+          ['facebook', 'github'],
+          // @ts-expect-error-this-line
```

Comment:
try `as unknown as xxx` to match type(in tests)

Reply:
Updated to `as ConnectorInstance[]`. PTAL.